client/src/components/pages/utils/teacherTap.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+const teacherTape = (num, data) => {
```

Comment:
same as above, pls put it back into the TeacherProfile component (this function is hardly reusable by anything else). Here are some options to make it more readable:
- you can extract each part into separate function:
```
renderAbout = (text) => (
      <div>
        <p className="teach-profile__desc">{text}</p>
      </div>
    );

....

if (activeContent === ABOUT) {
    renderAbout(data.about);
} else if (activeContent === CONTACT) {
     renderContact(data.contact);
} else if ....
```
- you can extract some parts into separate components. The Contact section would be a good candidate:
```
if (activeContent === CONTACT) {
    <TeacherContact contact={data.contact} />
}
```
If you do that, consider whether the new component can be reused from anywhere else. If so, put it into `components/common`, otherwise just keep it inside the parent component's module.
- you could extract the entire `render` function into a separate [presentational](https://flaviocopes.com/react-presentational-vs-container-components/) component. But that would be a bigger refactoring and is not really needed on this scale so I wouldn't go into that...